src/expr/src/relation/func.rs

```diff
@@ -3134,6 +3135,41 @@ fn regexp_extract(a: Datum, r: &AnalyzedRegex) -> Option<(Row, Diff)> {
     Some((Row::pack(datums), 1))
 }
 
+fn regexp_matches<'a, 'r: 'a>(
+    a: Datum<'a>,
+    r: &'r AnalyzedRegex,
+) -> impl Iterator<Item = (Row, Diff)> + 'a {
+    let regex = r.inner();
+    let a = a.unwrap_str();
+
+    let iter = regex.captures_iter(a).map(move |captures| {
+        let matches = captures
+            .iter()
+            // The first match is the *entire* match, we want the capture groups by themselves.
+            .skip(1)
+            .map(|m| Datum::from(m.map(|m| m.as_str())))
+            .collect::<Vec<_>>();
+
+        let mut row = Row::default();
+        let mut packer = row.packer();
```

Comment:
Here would be a good place to use `SharedRow`, to avoid reallocations in `try_push_array`.

Reply:
Can you check if I'm holding SharedRow correctly?

Reply:
@antiguru I might be missing an allocation somewhere, but how does `SharedRow` help? We need to return an iterator of owned `Row`s so allocating new `Row`s seems unavoidable?